### ISSUE TITLE
Syntax highlight broken if text contains ampersand

### ISSUE THREAD

**Issue by karlschwaier** [atom/language-html#181]
### Description

If a string contains an ampersand, the syntax highlight after that ampersand is broken.

![bildschirmfoto 2017-12-13 um 15 03 02](https://user-images.githubusercontent.com/2789472/33947989-19a8a0d0-e026-11e7-80e3-4ce81d4cb160.png)

### Versions

Atom 1.23.0
language-html 0.48.4

### Additional Information 

Similar to https://github.com/atom/language-html/issues/141.

**Comment by jonathanfleckenstein** [comment]
```
<iframe height="500px" width="500px"
  src="/v/?num=&height=&width=&autostart=true&repeat=false&force_dimensions=true&controlbar=false&disable_pause=true&disable_opening_frame=true&wmode=transparent&presentation_mode=product_preview&allow_full_screen=false"
  frameborder="0" allowFullScreen="false" webkitallowfullscreen="false" mozallowfullscreen="false"></iframe>
```
Also happens with this string, but only after the 4th ampersand.
Atom 1.22.1
language-html 0.48.2

**Comment by LtSorry** [comment]
It happens when an ampersand is immediately followed by any alphabetical character.

![image](https://user-images.githubusercontent.com/34553972/34012465-d9cd9530-e0e1-11e7-9abe-aa0cd031b77e.png)

vs.

![image](https://user-images.githubusercontent.com/34553972/34012485-eabbebb2-e0e1-11e7-88ac-d6cf56abe934.png)

Atom 1.23.1
language-html 0.48.4

**Comment by 50Wliu** [comment]
@jonathanfleckenstein your issue should be fixed in Atom 1.23.0.  I'm looking into the other issues now.

**Comment by jonathanfleckenstein** [comment]
@50Wliu Thank you, hope you're having a great Holiday!
